### PR TITLE
Add the default OpenPopupCallback implementations

### PR DESCRIPTION
**Pull request by serhii-kozionov**
We have the examples for the `OpenPopupCallback` implementation on the support portal. They should be kept up to date and not use the internal API. 
The purpose of this PR is to keep those examples up to date and allow adding them to the support portal automatically using the embedcode tool.
Additionally, all references to internal JxBrowser API are removed.

### CODE REVIEW

**Comment by serhii-kozionov** [comment]
@vladimir-ikryanov PTAL

**Comment by serhii-kozionov** [comment]
@vlad-lubenskyi PTAL

File: callbacks/src/main/java/javafx/DefaultOpenPopupCallback.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import com.teamdev.jxbrowser.os.Environment;
+import com.teamdev.jxbrowser.ui.Rect;
+import com.teamdev.jxbrowser.view.javafx.BrowserView;
+import javafx.application.Platform;
+import javafx.scene.Scene;
+import javafx.scene.layout.StackPane;
+import javafx.stage.Stage;
+
+/**
+ * The default {@link OpenPopupCallback} implementation for the Swing UI toolkit that creates and shows a new window
```
Comment (vlad-lubenskyi) [inline callbacks/src/main/java/javafx/DefaultOpenPopupCallback.java:37]: I think you meant `JavaFX` here

File: callbacks/src/main/java/javafx/DefaultOpenPopupCallback.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            StackPane root = new StackPane();
+            Scene scene = new Scene(root);
+            root.getChildren().add(view);
+            stage.setScene(scene);
+
+            updateBounds(stage, params.initialBounds());
+
+            stage.setOnCloseRequest(event -> {
+                if (Environment.isWindows()) {
+                    new Thread(browser::close).start();
```
Comment (vlad-lubenskyi) [inline callbacks/src/main/java/javafx/DefaultOpenPopupCallback.java:60]: Do we have it explained somewhere on our support portal? If yes, we should reference it here. If not, let's explain. This code may be an entry point to JxBrowser for a developer and it's important to explain such magic.

File: callbacks/src/main/java/swing/DefaultOpenPopupCallback.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                @Override
+                public void windowClosing(WindowEvent e) {
+                    invokeLater(browser::close);
+                }
+            });
+
+            updateBounds(frame, params.initialBounds());
+
+            browser.on(TitleChanged.class, event ->
+                    invokeLater(() ->
```
Comment (vlad-lubenskyi) [inline callbacks/src/main/java/swing/DefaultOpenPopupCallback.java:65]: It fits on a single line.

File: callbacks/src/main/java/swt/DefaultOpenPopupCallback.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+import com.teamdev.jxbrowser.ui.Rect;
+import com.teamdev.jxbrowser.view.swt.BrowserView;
+import org.eclipse.swt.SWTException;
+import org.eclipse.swt.layout.FillLayout;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.swt.widgets.Widget;
+
+/**
+ * The default {@link OpenPopupCallback} implementation for the Swing UI toolkit that creates and shows a new window
```
Comment (vlad-lubenskyi) [inline callbacks/src/main/java/swt/DefaultOpenPopupCallback.java:37]: Same thing, `SWT`.

File: callbacks/src/main/java/swt/DefaultOpenPopupCallback.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+                        asyncExec(shell, browser::close);
+                    }
+                });
+
+                browser.on(TitleChanged.class, event ->
+                        asyncExec(shell, () -> shell.setText(event.title())));
+
+                browser.on(BrowserClosed.class, event ->
+                        asyncExec(shell, shell::dispose));
+
```
Comment (vlad-lubenskyi) [inline callbacks/src/main/java/swt/DefaultOpenPopupCallback.java:67]: No need for empty lines here.

File: callbacks/src/main/java/swt/DefaultOpenPopupCallback.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+    private static void asyncExec(Widget widget, Runnable doRun) {
+        try {
+            widget.getDisplay().asyncExec(() -> {
+                if (!widget.isDisposed()) {
+                    doRun.run();
+                }
+            });
+        } catch (SWTException ignore) {
```
Comment (vlad-lubenskyi) [inline callbacks/src/main/java/swt/DefaultOpenPopupCallback.java:103]: This is a strange code to be in the example. If I were a client I'd read it as "we can throw some exceptions inside, just ignore them".  It smells. I know that it's not far from the truth and I regularly saw the exceptions "*** is already disposed" during the browser closing. 

I think this detail is redundant in this example, let's go with the standard `Display.asyncExec` everywhere.

File: troubleshooting/src/main/java/RedirectLoggingToFile.java [hunk 7]
```diff
@@ -39,8 +40,12 @@
 public final class RedirectLoggingToFile {
 
     public static void main(String[] args) {
-        // Redirect Browser log messages to jxbrowser-browser.log
-        Path loggingDir = Files.createTempDir("jxbrowser-logs");
+        Path loggingDir;
+        try {
+            loggingDir = Files.createTempDirectory("jxbrowser-logs");
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to create temporary directory", e);
```
Comment (vlad-lubenskyi) [inline troubleshooting/src/main/java/RedirectLoggingToFile.java:47]: An article is missing.

File: troubleshooting/src/main/java/RedirectLoggingToFile.java [hunk 8]
```diff
@@ -39,8 +40,12 @@
...
+        }
         Path loggingFile = loggingDir.resolve("jxbrowser-browser.log");
```
Comment (vlad-lubenskyi) [inline troubleshooting/src/main/java/RedirectLoggingToFile.java:49]: Why `jxbrowser-browser`, btw?

**Review by vlad-lubenskyi — CHANGES_REQUESTED** [review]

File: callbacks/src/main/java/swt/DefaultOpenPopupCallback.java (hunk 6 above)
Comment (serhii-kozionov) [reply callbacks/src/main/java/swt/DefaultOpenPopupCallback.java:103]: We mustn't call methods of the disposed `Shell`. I have removed the `try-catch` block.

File: callbacks/src/main/java/javafx/DefaultOpenPopupCallback.java (hunk 2 above)
Comment (serhii-kozionov) [reply callbacks/src/main/java/javafx/DefaultOpenPopupCallback.java]: This code is useless starting from 7.10

File: troubleshooting/src/main/java/RedirectLoggingToFile.java (hunk 8 above)
Comment (serhii-kozionov) [reply troubleshooting/src/main/java/RedirectLoggingToFile.java:49]: Hard to say. I changed to `jxbrowser.log`.

**Comment by serhii-kozionov** [comment]
@vlad-lubenskyi PTAL

File: troubleshooting/src/main/java/RedirectLoggingToFile.java (hunk 8 above)
Comment (vladimir-ikryanov) [reply troubleshooting/src/main/java/RedirectLoggingToFile.java]: It legacy from JxBrowser 6 https://jxbrowser.support.teamdev.com/support/solutions/articles/9000013125-logging

**Review by vlad-lubenskyi — APPROVED** [review]

**Pull request merged**